pkg/aws/rds/rds.go

```diff
@@ -48,15 +48,40 @@ func NewClient(ctx context.Context, awsConfig *aws.Config, region string) (*RDS,
 }
 
 // CreateDBInstanceWithContext
-func (r RDS) CreateDBInstance(ctx context.Context, storage int64, instanceClass, instanceID, engine, username, password string, sgIDs []string) (*rds.CreateDBInstanceOutput, error) {
-	dbi := &rds.CreateDBInstanceInput{
-		AllocatedStorage:     &storage,
-		DBInstanceIdentifier: &instanceID,
-		VpcSecurityGroupIds:  convertSGIDs(sgIDs),
-		DBInstanceClass:      &instanceClass,
-		Engine:               &engine,
-		MasterUsername:       &username,
-		MasterUserPassword:   &password,
+func (r RDS) CreateDBInstance(ctx context.Context, storage *int64, instanceClass, instanceID, engine, username, password string, sgIDs []string, publicAccess *bool, restoredClusterID *string, dbSubnetGroup string) (*rds.CreateDBInstanceOutput, error) {
+	var dbi *rds.CreateDBInstanceInput
+
+	// check if the instance is being restored from an existing cluster
+	switch {
+	case restoredClusterID != nil && publicAccess != nil:
+		dbi = &rds.CreateDBInstanceInput{
+			DBClusterIdentifier:  restoredClusterID,
+			DBInstanceClass:      &instanceClass,
+			DBInstanceIdentifier: &instanceID,
+			DBSubnetGroupName:    &dbSubnetGroup,
+			Engine:               &engine,
+			PubliclyAccessible:   publicAccess,
+		}
+	case restoredClusterID != nil && publicAccess == nil:
+		dbi = &rds.CreateDBInstanceInput{
+			DBClusterIdentifier:  restoredClusterID,
+			DBInstanceClass:      &instanceClass,
+			DBInstanceIdentifier: &instanceID,
+			DBSubnetGroupName:    &dbSubnetGroup,
+			Engine:               &engine,
+		}
+	default:
+		// if not restoring from an existing cluster, create a new instance input
+		dbi = &rds.CreateDBInstanceInput{
+			AllocatedStorage:     storage,
+			DBInstanceIdentifier: &instanceID,
+			VpcSecurityGroupIds:  convertSGIDs(sgIDs),
+			DBInstanceClass:      &instanceClass,
+			Engine:               &engine,
+			MasterUsername:       &username,
+			MasterUserPassword:   &password,
+			PubliclyAccessible:   publicAccess,
+		}
```

Comment:
can we initialise the struct with common fields, outside of the switch and then in case just set the fields that are specific to that case?

Reply:
Yes, init the struct with *common* fields before switch and then in case statements just add fields respective to that case.

Reply:
Yeah did that. PTAL.